src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -22,17 +22,16 @@ internal class JsonClaimSet
 
         internal JsonClaimSet(JsonDocument jsonDocument)
         {
-            RootElement = jsonDocument.RootElement;
+            RootElement = jsonDocument.RootElement.Clone();
```

Comment:
We might want to add a comment here explaining, just in case this code lives longer than we expect.

Reply:
```suggestion
            // This method is assuming ownership of the JsonDocument, which is backed by one or more ArrayPool arrays.
            // We need to dispose of it to avoid leaking arrays from the pool.  To achieve that, we clone the root element,
            // which will result in a new JsonElement being created that's not tied to the original and that's not backed by
            // ArrayPool memory, after which point we can dispose of the original to return the array(s) to the pool.
            RootElement = jsonDocument.RootElement.Clone();
```